### PR TITLE
Christopher foster

### PR DESCRIPTION
**Pull request by catslanderer**

### CODE REVIEW

File: enhancing/enhancer.js [hunk 1]
```diff
@@ -2,19 +2,25 @@ module.exports = {
...
 }
 
 function fail(item) {
-  return { ...item };
+  const enhancement =
+    item.enhancement > 16 ? item.enhancement - 1 : item.enhancement;
+  let durability =
+    item.enhancement < 15 ? item.durability - 5 : item.durability - 10;
+  durability = durability < 0 ? 0 : durability;
+  return { ...item, enhancement, durability };
```
Comment (DaftBeowulf) [inline enhancing/enhancer.js]: curveball time--should you account for if a durability would go below 0?

File: enhancing/enhancer.js [hunk 2]
```diff
@@ -2,19 +2,25 @@ module.exports = {
...
+  return { ...item, enhancement };
 }
 
 function fail(item) {
-  return { ...item };
+  const enhancement =
+    item.enhancement > 16 ? item.enhancement - 1 : item.enhancement;
+  let durability =
+    item.enhancement < 15 ? item.durability - 5 : item.durability - 10;
+  durability = durability < 0 ? 0 : durability;
```
Comment (DaftBeowulf) [inline enhancing/enhancer.js:18]: good job accounting for this edge case :+1:

File: enhancing/enhancer.spec.js [hunk 3]
```diff
@@ -1,2 +1,77 @@
 const enhancer = require('./enhancer.js');
 // test away!
+
+it('item => new item with durability === 100', () => {
```
Comment (DaftBeowulf) [inline enhancing/enhancer.spec.js:4]: for this size app this isn't wrong, but it's always a good idea to wrap tests dealing with the same function in the same `describe` block, and arguably even all function `describe` blocks should be wrapped in a single `describe` block for the file they're all imported from to be more readable and keep things separated--check how Luis did it in the guided demo today to see an example

File: enhancing/enhancer.spec.js [hunk 4]
```diff
@@ -1,2 +1,77 @@
...
+  const received = {
+    name: 'Sword of Regret',
+    enhancement: 12,
+    durability: 28
+  };
+  const expected = {
+    name: 'Sword of Regret',
+    enhancement: 12,
+    durability: 100
+  };
```
Comment (DaftBeowulf) [inline enhancing/enhancer.spec.js:14]: this pattern of being extra-explicit for each object you set up for your test is very declarative, but you're going to end up repeating yourself and extending your line count quite a bit. once you feel comfortable with the jest matcher/expected syntax, start cutting down by either declaring the object globally and resetting it with a `beforeEach` method, or at the very least only check the properties you actually want to test for. for this first one for instance, instead of matching the complete objects with `toEqual`, just check the `repair(received).durability` and match it `toBe(100)` to save yourself some time and be even more explicit not about what you're passing in, but what specific property you're testing for

File: enhancing/enhancer.spec.js [hunk 5]
```diff
@@ -1,2 +1,77 @@
...
+  expect(enhancer.repair(received)).toEqual(expected);
+});
+
+it('item => new item with enhancement + 1 (up to a max of 20)', () => {
```
Comment (DaftBeowulf) [inline enhancing/enhancer.spec.js:18]: there's nothing wrong with this, but one of the nice things about the jest `it` pattern is that it lends itself to very English-readable test descriptions--it makes it slightly more readable if this were to say `it('increments the item's enhancement by 1 (max 20)` for example. another reason while it's a good idea to also wrap your functions' `it` statements in a `describe` wrapper is you can say exactly which function you're testing here, instead of having to read it in the actual test below, ex: `describe("succeed()", ()=>{
it("should do stuff", ()=> {
test
})
})`

**Review by DaftBeowulf — COMMENTED** [review]
Good job with the MVP Chris, although check my comments since you can make some improvements that would both make your test more readable and more concise
